L1Trigger/Phase2L1ParticleFlow/test/testOutputFiles.cpp

```diff
@@ -0,0 +1,452 @@
+
+			if (debug) {
+				std::cout << "bset_string_ = " << bset_string_ << std::endl;
+				std::cout << "\tap_uint<14> hwPt = " << std::flush; for (int i=14-1; i>=0; i--) {std::cout << bset_[i] << std::flush;} std::cout << std::endl;
+				std::cout << "ap_int<12> hwVtxPhi = " << std::flush; for (int i=12-1; i>=0; i--) {std::cout << bset_[i+15] << std::flush;} std::cout << std::endl;
+				std::cout << "ap_int<16> hwVtxEta = " << std::flush; for (int i=16-1; i>=0; i--) {std::cout << bset_[i+27] << std::flush;} std::cout << std::endl;
+				std::cout << "\thwPt (int) = " << hwPt.to_int() << std::endl;
+				std::cout << "\thwVtxPhi (int) = " << hwVtxPhi.to_int() << std::endl;
+				std::cout << "\thwVtxEta (int) = " << hwVtxEta.to_int() << std::endl;
+				std::cout << "\thwVtxPt_f (float) = " << hwVtxPt_f << std::endl;
+				std::cout << "\thwVtxPhi_f (float) = " << hwVtxPhi_f << std::endl;
+				std::cout << "\thwVtxEta_f (float) = " << hwVtxEta_f << std::endl;
+			}
+
+			if (bset_.any()) {
+				ntrackstotal++;
+				tracks_coe[std::make_pair(isector,iline)] = makeTLorentzVectorPtEtaPhiE(hwVtxPt_f,hwVtxEta_f,hwVtxPhi_f,hwVtxPt_f);
+				//if (print) printf("\t\t Track %u (pT,eta,phi): (%.4f,%.4f,%.4f)\n", it, hwPt_f, hwVtxEta_f, hwVtxPhi_f);
+			}
+		}
+
+		// remove the trailing character
+		bset_string_.resize(2);
+		cfile_.read(&bset_string_[0],2);
+		if (debug && !skip) std::cout << "bset_string_ = " << bset_string_ << std::endl;
+		if (bset_string_!=",\n" && bset_string_!=";\n") {
+			std::cerr << "ERROR::testDumpFile::get_tracks_from_coe_file Something went wrong reading line " << 11+iline << " of the COE file"  << std::endl
+					  << "\tThe line should have ended with \',<newline>\' or \';<newline>\', but instead ended with \'" << bset_string_ << "\'" << std::endl;
+			assert(bset_string_!="," || bset_string_!=";");
+		}
+
+	}
+	for (unsigned int is = 0; is < N_IN_SECTORS; ++is) {
+		std::vector<SectorTrackIndex> tracks_in_sector;
+		findAllInRegion<SectorTrackIndex,TLorentzVector*,int>(tracks_in_sector,tracks_coe,is);
+		if (print) printf("\tRead region %u [%0.2f,%0.2f] with %lu tracks\n", is, SECTOR_BOUNDARIES[is], SECTOR_BOUNDARIES[is+1], tracks_in_sector.size());
+		for (unsigned int it=0; it<tracks_in_sector.size(); it++) {
+			if (print) printf("\t\t Track %u (pT,eta,phi): (%.4f,%.4f,%.4f)\n", it, tracks_coe[tracks_in_sector[it]]->Pt(), tracks_coe[tracks_in_sector[it]]->Eta(), tracks_coe[tracks_in_sector[it]]->Phi());
+		}
+	}
+
+	if (print) {
+		printf("\t================================= \n");
+		printf("\tTotal tracks %u \n\n", ntrackstotal);
+	}
+
+	return tracks_coe;
+}
+
+std::ifstream& GotoLine(std::ifstream& file, unsigned int num){
+	file.seekg(std::ios::beg);
+	for(unsigned int i=0; i < num - 1; ++i){
+		file.ignore(std::numeric_limits<std::streamsize>::max(),'\n');
+	}
+	return file;
+}
+
+bool compare_lv_with_tolerance(TLorentzVector a, TLorentzVector b, std::vector<float> tolerance = {0,0,0}) {
```

Comment:
can't it be const std::vector<float> & ?
also, here it's initialized with 3 values but below you use tolerance[4]

Reply:
You are right. I will fix this in my next commit.